src/boot/bootctl-uki.c

```diff
@@ -0,0 +1,96 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "bootctl.h"
+#include "bootctl-uki.h"
+#include "fd-util.h"
+#include "parse-util.h"
+#include "pe-header.h"
+
+#define DOS_FILE_MAGIC "MZ"
+#define PE_FILE_MAGIC  "PE\0\0"
+
+static const uint8_t name_osrel[8] = ".osrel";
+static const uint8_t name_linux[8] = ".linux";
+static const uint8_t name_initrd[8] = ".initrd";
+
+static struct PeSectionHeader* pe_sections(FILE *uki, uint16_t *n)
+{
+        struct PeSectionHeader *sections;
+        struct DosFileHeader dos;
+        struct PeHeader pe;
+        uint16_t scount;
+        off_t soff;
```

Comment:
off_t is so yuck. I'd really use `uint64_t` to ensure this means the same thing no matter what.